torch/lib/TH/THSize.h

```diff
@@ -0,0 +1,13 @@
+#ifndef TH_SIZE_INC
+#define TH_SIZE_INC
+
+#include "THGeneral.h"
+#include <stddef.h>
+
+// THTensor functions that would work on a THSize if we had such a class in C++,
+// i.e. THTensor functions that depend only on the shape of the tensor, not the type.
```

Comment:
THTensor functions that depend on the shape, but not type are all `THLongStorage` functions, right? That's what we've used for `inferSize` etc. I think it's a better place for it